src/foundry/foundry.js/pixi/containers/placeableObjects/token.d.ts

```diff
@@ -500,43 +433,26 @@ declare global {
 
     /**
      * Extend the PlaceableObject.rotate method to prevent rotation if the Token is in the midst of a movement animation
+     * @returns Actually a Promise<void>
      */
     rotate(angle: number, snap: number): Promise<this>;
 
     /** @override */
-    // TODO: Replace any with TokenDocument
-    protected _onCreate(options?: DeepPartial<any>, userId?: DocumentModificationOptions): void;
+    protected _onCreate(
+      options: InstanceType<ConfiguredDocumentClass<typeof TokenDocument>>['data']['_source'],
+      userId: DocumentModificationOptions
+    ): void;
 
     /** @override */
-    // TODO: Replace any with TokenDocument
     protected _onUpdate(
-      data?: DeepPartial<any>,
+      data?: DeepPartial<InstanceType<ConfiguredDocumentClass<typeof TokenDocument>>['data']['_source']>,
       options?: DocumentModificationOptions & { animate?: boolean },
```

Comment:
where does the `animate` come from?

Reply:
From the body of the method: `const animating = positionChange && (options.animate !== false);`